test/Deletion.hs

```diff
@@ -15,71 +16,43 @@ tests = describe "Delete" $ do
     it' "Tree is still balanced after delete" $ \list -> do
         tree  <- AVL.fromList list :: StorageMonad M
         trees <- scanM (AVL.deleteWithNoProof . fst) tree list
-        yes   <- allM  (AVL.isBalancedToTheLeaves)   trees
+        yes   <- and <$> for trees AVL.isBalancedToTheLeaves
 
         for_ trees $ AVL.isBalancedToTheLeaves
 
         return yes
 
-    it' "Deletion deletes" $ \list -> do
-        if length list == 0
-        then do
-            return True
-
-        else do
-            let (k, _) : _  = list
-            tree  <- AVL.fromList list :: StorageMonad M
-            tree1 <- AVL.deleteWithNoProof k tree
-            list' <- AVL.toList tree1
-            let diff = uniqued list \\ list'
-            return $ length diff == 1 && fst (head diff) == k
+    it' "dds Deletion deletes" $ \(k, v, list) -> do
```

Comment:
Please, don't forget to remove your pattern shortcuts from the descriptions. Also, you can just use `focus` temporarily if you need to debug one particular test.

Reply:
Fix'd